cpp/cmake/Dependencies.cmake

```diff
@@ -16,6 +16,32 @@
 
 include(ExternalProject)
 
+##############################################################################
+# - raft - (header only) -----------------------------------------------------
+
+# Only cloned if RAFT_PATH env variable is not defined
+
+if(DEFINED ENV{RAFT_PATH})
```

Comment:
@dantegd @JohnZed @cjnolet For now the `Dependencies.cmake` changes are fine since we have agreed to keep raft as a header-only library. What if that changes in future and we might have to build binaries? Then this if-else block would become difficult to support, no?

Reply:
If we need to build binaries, then we will need to reopen design discussions and overall will have big implications. It is agreed right now with the cuGraph team to keep it header only, if that changes then this will need to change, and might introduce the need for a conda package distribution and other matters

Reply:
But thinking about it, I don’t think that, if a binary is needed, it would need a massive change. Mainly having to modify the `CMakeLists` in RAFT, then that gets included by libcugraph/libcuml and then (presumably) statically linking the libraft generated, no?

Reply:
Good point. How about doing something similar to what you've been doing to share RAFT's python code? In other words, we expose a `RAFT_SRC_FILES` cmake variable and it is the responsibility of cuml/cugraph to include those during their build?

Reply:
actually there might be an even better/more sustainable way, since `ExternalProject_Add` can be used to trigger build processes of local folders (as opposed to just downloading/git cloning). This would allow the cmake code to be the same for working with a local or cloned RAFT, as well as triggering the build from cuML's cmake, let me do the change to see how it looks

Reply:
@teju85 now both paths are using `ExternalProject_Add` so that if we need to eventually configure raft, or build raft (and statically link it or include it in the libcuml package) in the future, the build infra would support that. 

Couldn't pull it from the if since the `DOWNLOAD_COMMAND  ""` parameter overrides the git cloning no matter what :/

Reply:
This is atleast better than the previous one. Thanks Dante for looking into this..